### ISSUE TITLE
[com_fields] Contact-Mail Custom field issue after joomla 3.8.3

### ISSUE THREAD

**Issue by nader24** [joomla/joomla-cms#20195]
Hello,

I had 2 contacts on my page and i made 2 different catagory for mail-custom-fields and made also different custom field for each catagory and contact it was everything ok on joomla 3.8.3 after update to  joomla 3.8.5 and more the custom field appear  on all of contacts and not just on selected field categories.

i tried to use a new instal joomla but it has the same problem i hope you can to help me to solve this issue .

thank you
regards
N.K

**Comment by ghost** [comment]
Please ask help on the [forums](https://forum.joomla.org/). This repository concerns in first Place Joomla-Core coding. For this Reason closing this Issue, thanks.

**Comment by ReLater** [comment]
@franz-wohlkoenig 
Could you please reopen this issue. I can confirm it for current Staging Joomla 3.8.8-dev.

- Create 2 contact categories (Test-Cat-1 and Test-Cat-2).
- Create 2 contacts:
1) Test-Contact-1 with category Test-Cat-1.
2) Test-Contact-2 with category Test-Cat-2.
- Create 2 com_fields fields of type "Mail".
1) Test-Cat-1 field. Select Category: Test-Cat-1.
2) Test-Cat-2 field. Select Category: Test-Cat-2.
- Create 2 menu items of type "single contact":
1) for contact Test-Contact-1.
2) for contact Test-Contact-2.

- Go to frontend and open the menu items.
- In both contact forms are ALL fields visible.

![18-04-_2018_21-18-20](https://user-images.githubusercontent.com/20780646/38953178-2fa3e8aa-434e-11e8-96b8-a5c2b739f4e8.jpg)

**Comment by ghost** [comment]
reopened as stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20195">issues.joomla.org/tracker/joomla-cms/20195</a>.</sub>

**Comment by ReLater** [comment]
Thank you!

**Comment by ReLater** [comment]
Some investigations:
We had this issue already in the past. @laoneo correted it here:
https://github.com/joomla/joomla-cms/pull/16666/files
https://github.com/joomla/joomla-cms/pull/16666

In Joomla 3.8.4 these lines were added by @sandewt 
https://github.com/joomla/joomla-cms/pull/17743/files
https://github.com/joomla/joomla-cms/pull/17743

From the point of logic both fixes look somehow conflictive.

**Comment by Quy** [comment]
In the meantime, under each field, under `Options` tab, set `Display When Read-Only` to No.

**Comment by ReLater** [comment]
> In the meantime, under each field, under Options tab, set Display When Read-Only to No.

That my fields are only read-only is just because of my laziness concerning writing testing instructions ;-)
Makes no difference. The issue was also reported in German forum where the fields are editable.

**Comment by ReLater** [comment]
Sorry for pushing but this is a serious bug for some Joomla users!
Could we get here comments please from @laoneo and/or @sandewt 
My problem is that I don't understand https://github.com/joomla/joomla-cms/pull/17743

**Comment by laoneo** [comment]
I can have a look next week.

**Comment by sandewt** [comment]
>Sorry for pushing but this is a serious bug for some Joomla users!

@ReLater very clear instruction.

Tested with: `Joomla! 3.8.8-dev Development`

File + location: `joomla-cms-staging\components\com_contact\views\contact\view.html.php`

If you change this code in the rules 75 ...

```
if (empty($item->catid))
{
	$app->setUserState('com_contact.contact.data', array('catid' => $item->catid));
}
```
back to one rule

```
$app->setUserState('com_contact.contact.data', array('catid' => $item->catid));
```

The result is 1 field, as you wish. See the image.


![screen shot 2018-04-22 at 08 54 26](https://issues.joomla.org/uploads/1/f9ae3cfe948ccc4715d51bd396efe697.png)

However, `Contact form loses data after postback with error` is then an issue (#17743) again.
Honestly, I do not know a solution yet.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20195">issues.joomla.org/tracker/joomla-cms/20195</a>.</sub>

**Comment by degobbis** [comment]
Why do you try to set the userState for com_contact.contact.data if catid is empty?

**Comment by sandewt** [comment]
@degobbis 
May be not the best argument. Because it worked and the test results were positive.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20195">issues.joomla.org/tracker/joomla-cms/20195</a>.</sub>

**Comment by ghost** [comment]
closed as having Pull Request #20214

**Comment by degobbis** [comment]
@ReLater  @sandewt 
Please test #20214 

Thanks
